apps/snitch_core/test/data/model/country_test.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Snitch.Data.Model.CountryTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  alias Snitch.Data.Model.Country
+  import Snitch.Factory
+
+  setup :countries
+
+  describe "Countries" do
+
+    test "for a single country ", %{countries: countries} do
+      country = countries |> List.first()
+      country_list = Country.formatted_list()
+      assert country_list == [{country.name, country.id}]
+    end
+
+    test "for getting records by a particular key ", %{countries: countries} do
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: country.numcode})
+      assert country_list == country
+    end
+
+    test "fails, Why? value not present for a given key", %{countries: countries} do
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: "842"})
```

Comment:
We can test without fetching the first country from the list and with the actual value

```elixir
assert Country.get(%{numcode: "842"}) == nil
```